exercises/concept-wip/arys-amazing-lasagna/.meta/src/exemple.cr

```diff
@@ -0,0 +1,16 @@
+class Lasagna
+  EXPECTED_MINUTES_IN_OVEN      = 40
+  PREPARATION_MINUTES_PER_LAYER =  2
+
+  def self.remaining_minutes_in_oven(actual_minutes_in_oven)
+    EXPECTED_MINUTES_IN_OVEN - actual_minutes_in_oven
+  end
+
+  def self.preparation_time_in_minutes(layers)
+    layers * PREPARATION_MINUTES_PER_LAYER
+  end
+
+  def self.total_time_in_minutes(number_of_layers, actual_minutes_in_oven)
+    preparation_time_in_minutes(number_of_layers) + actual_minutes_in_oven
+  end
+end
```

Comment:
I think that for a student it would be easier to have these as regular methods, not using the `self.` notation:

```suggestion
class Lasagna
  EXPECTED_MINUTES_IN_OVEN      = 40
  PREPARATION_MINUTES_PER_LAYER =  2

  def remaining_minutes_in_oven(actual_minutes_in_oven)
    EXPECTED_MINUTES_IN_OVEN - actual_minutes_in_oven
  end

  def preparation_time_in_minutes(layers)
    layers * PREPARATION_MINUTES_PER_LAYER
  end

  def total_time_in_minutes(number_of_layers, actual_minutes_in_oven)
    preparation_time_in_minutes(number_of_layers) + actual_minutes_in_oven
  end
end
```

This might not be the idiomatic solution once you know a bit more of Crystal, but it does reduce the number of things the student needs to know.
It is totally fine for earlier exercises to use non-idiomatic solutions, as the student will inevitably not be familiar with all of the language's constructs.

Once the `self.` bit is removed, the corresponding explanation in the `introduction.md`/`about.md` files can also be removed.
You could then have a separate exercise that teaches about class methods.